build_tools/index_generation_s3_tar.py

```diff
@@ -104,18 +136,31 @@ def generate_index_s3(bucket_name, region_name='us-east-2', prefix=''):
     local_path = "index.html"
     with open(local_path, "w") as f:
         f.write(html_content)
-    
-    print(f"index.html generated successfully for bucket '{bucket_name}'. File saved as {local_path}")
 
-    # Upload index.html back to S3
+    print(
+        f"index.html generated successfully for bucket '{bucket_name}'. File saved as {local_path}"
+    )
+
     try:
-        s3.upload_file(local_path, bucket_name, 'index.html', ExtraArgs={'ContentType': 'text/html', 'ACL': 'public-read'})
-        print(f"index.html successfully uploaded to bucket '{bucket_name}' with public-read ACL.")
+        s3.upload_file(
+            local_path,
+            bucket_name,
+            "index.html",
+            ExtraArgs={"ContentType": "text/html"},
+        )
+        print(f"index.html successfully uploaded to bucket '{bucket_name}'.")
     except ClientError as e:
         print(f"Failed to upload index.html to bucket '{bucket_name}': {e}")
 
+
 if __name__ == "__main__":
-    bucket_name = 'therock-nightly-tarball'
-    region_name = 'us-east-2'
-    prefix = ''
-    generate_index_s3(bucket_name, region_name, prefix)
+    parser = argparse.ArgumentParser(
+        description="Generate index.html for S3 bucket tar.gz files."
+    )
+    parser.add_argument("--bucket", required=True, help="S3 bucket name")
+    args = parser.parse_args()
+
+    region_name = "us-east-2"
+    prefix = ""
+
+    generate_index_s3(args.bucket, region_name, prefix)
```

Comment:
Can you use keyword arguments here? If all arguments are the same type (bool, string, etc.) it is very easy to get the argument ordering wrong.

```suggestion
    region_name = "us-east-2"
    prefix = ""

    generate_index_s3(bucket_name=args.bucket, region_name=region_name, prefix=prefix)
```

(might need to be formatted with pre-commit)